src/system/CScopedLock.h

```diff
@@ -26,12 +26,12 @@ namespace tigl
 
 class CMutex;
 
-class CScopedLock
+class CScopedLock // TODO: replace by std::lock_guard
```

Comment:
Please remove first TODO. As long as we have to stick with C++98, this is not going to happen.

What we could do is, if C++11 is available, CScopedLock is a simple typedef to std::lock_guard